### PR TITLE
filter sequences for blooms

### PR DESCRIPTION
**Pull request by wasade**
@ElDeveloper @antgonza @mortonjt @josenavas @squirrelo, round 2 is up.

### CODE REVIEW

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+Early in the American Gut Project, it was observed that some organisms bloomed likely as a result of increased shipping time and delay between when samples were collected and when they were put on ice (more detail can be found [here](http://americangut.org/?page_id=277)). The purpose of this notebook is to apply the filter developed in order to bioinformatically subtract these observed bloom sequences from fecal samples. It is important to apply this filter when combining data with the American Gut as to remove a potential study-effect bias as all fecal data in the American Gut has had this filter applied. The specific steps covered are:
+
+* Filter demultiplexed sequence data to only fecal samples
+* Determine what sequences in the fecal samples recruit to the observed bloom sequences
+* Remove the recruited bloom sequences from the demultiplexed sequence data
+* Trim the filtered demultiplexed data (to reduce study bias when combining with short reads, such as the data in [Yatsunenko et al 2012](http://www.nature.com/nature/journal/v486/n7402/abs/nature11053.html))
```
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Should this be in a separate md? Note that this is not part of the bloom filtering ...

Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Good catch, copy pasta. Trimming is done in the pick OTUs step

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+
+The filtering is only intended to be applied to fecal data. As such, this notebook allows you to describe what metadata column and value to use so that only fecal samples are used.
```
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Where do you define that? I guess that should be just after this line.

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md (hunk 1 above)
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: I want pasta, it's almost lunch time ...

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+...                  --valid_states $_fecal_states
+```
+
+The next thing we need to do is setup the parameters for SortMeRNA, which is the method we'll use to compare all the input data to our reference of bloom sequences.
+
+```python
+>>> _params_file = agu.get_path('sortmerna_pick_params.txt')
+>>> with open(_params_file, 'w') as f:
+...     f.write("pick_otus:otu_picking_method sortmerna\n")
+...     f.write("pick_otus:similarity 0.97\n")
```
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Isn't this the default?

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+As the data have now been filtered for blooms, we can now trim the reads back to 100nt to minimize a potential study effect when combining with the Global Gut.
+
+```python
+>>> with open(filtered_sequences) as in_, open(filtered_sequences_100nt, 'w') as out:
+...     agu.trim_fasta(in_, out, 100)
+```
+
+Finally, let's do a quick sanity check that we have sequence data and we'll also dump out summary information about how many reads per sample recruited to the blooms.
```
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: I get that this is a sanity check but > than zero doesn't say much. What about testing same number of seqs or something like that ...

**Comment by antgonza** [comment]
A few comments ... nothing major.

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md (hunk 4 above)
Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: What would the expected number be?

**Comment by wasade** [comment]
Thanks!

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md (hunk 4 above)
Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: As many as agu.trim_fasta writes?

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md (hunk 1 above)
Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: this is actually done in this notebook at the end. i dont see the benefit in a notebook dedicated to trimming though...

Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Yes, saw that. Perhaps adding to the main description. Note that it talks about the bloom filtering but not about the length.

File: ipynb/primary-processing/2-filter_sequences_for_blooms.md (hunk 4 above)
Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: What I don't like about this is the extensive amount of time it'll take to walk over the 2 x 30GBs of sequence data.

Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: Up to you, can make the change if you'd like

Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: What about removing the > than zero tests?

Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: So not performing any sanity checking?

Comment (antgonza) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: I guess the biom summary could suffice. Again, not strong feelings about this but IMOO a sanity check of file > than 0 is not really sanity, right?

Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: The `observed_blooms_biom` file is not used anywhere after this notebook. Checking that a file is nonzero tells you it has data, which is what we expect. We do not have expectations on the number of sequence counts as that will change over time, although we could assert that there are at least 10000 sequences but that also doesn't really tell you much. Testing if the number of reads is the same between these files tells you a little more than the nonzero test but all it says is that the data appear to be fasta-like. However, we already expect that as we assume QIIME's `filter_fasta.py` script works and similarly, we expect `trim_fasta` to work as it is both extremely simple and covered by an exercised unit test. 

If the files that result from this notebook are empty, it'll cause confusion on the pick OTUs phase, so the goal is to identify possible issues as early as is feasible.

Comment (wasade) [inline ipynb/primary-processing/2-filter_sequences_for_blooms.md]: And actually, testing if the number of reads is the same would still pass if both files were empty

**Comment by mortonjt** [comment]
I don't see any immediate problems.  Is this ready to merge?

**Comment by wasade** [comment]
If @antgonza gives a :+1:

**Comment by wasade** [comment]
thanks!

**Pull request merged**